app/src/main/java/org/mozilla/fenix/utils/Settings.kt

```diff
@@ -111,6 +113,50 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         default = 0
     )
 
+    private var experimentDefaultBrowserLastTappedExperiment by stringPreference(
+        appContext.getPreferenceKey(R.string.pref_key_experiment_default_last_tapped_experiment),
+        default = ""
+    )
+
+    private var experimentDefaultBrowserLastTappedTimeInMillis by longPreference(
+        appContext.getPreferenceKey(R.string.pref_key_experiment_default_last_tapped_time_in_millis),
+        default = -1
+    )
+
+    /*
+    * As there is not a consistent way to across OS versions to determine that the default browser
+    * was updated as result of tapping of one of the experiment, we are using some heuristics,
+    * to determine it.
+    * 1) The user has previously tapped one of the experiments.
+    * 2) The actual default browser it's Firefox.
+    * 4) The default browser was updated in an interval of 5 minutes of the last tap on
+    * an experiment.
+    * */
+    @Suppress("MagicNumber")
+    fun wasFirefoxSetAsDefaultByExperiment(experiment: Event.ExperimentDefaultBrowser.Type): Boolean {
+        val minuteInMS = 60 * 1000
+        val maxAllowedTime = 5 * 1000 // Max time between the last intent of setting a default browser
+        val browsers = BrowsersCache.all(appContext)
+        val wasSameExperiment = experimentDefaultBrowserLastTappedExperiment == experiment.name
+        val wasRecentlyUpdated =
+            experimentDefaultBrowserLastTappedTimeInMillis < Date().time - maxAllowedTime * minuteInMS
+
+        // We want to determine if the previous intent of setting a default browser was successful
+        // and was produced by this experiment.
+
+        return browsers.isDefaultBrowser && wasSameExperiment && wasRecentlyUpdated
+    }
+
+    fun recordIntentOfSettingDefaultBrowserExperiment(experiment: Event.ExperimentDefaultBrowser.Type) {
+        experimentDefaultBrowserLastTappedExperiment = experiment.name
+        experimentDefaultBrowserLastTappedTimeInMillis = Date().time
+    }
+
+    fun clearDefaultBrowserExperimentValues() {
+        experimentDefaultBrowserLastTappedExperiment = ""
+        experimentDefaultBrowserLastTappedTimeInMillis = -1
+    }
```

Comment:
@tdsmith I as shared on the comments of the google [docs](https://docs.google.com/document/d/1fcG8V54gdEywQcSRJTsBNgzOM3EQWnmPF7jqElxpzcs/edit ), we don't have reliable to detect when an experiment let to users to change their default browser, we are using a set of heuristics that help us to detect which experiment let user to switch their default browser.